hw/ip/spi_host/doc/_index.md

```diff
@@ -0,0 +1,633 @@
+The `cs_n` line can then be deasserted by either issuing another command after clearing the {{< regref "CONFIG.CSAAT" >}} field, issuing a command to a different CSID, or simply resetting the core FSM via the {{< regref "CONTROL.RST_FSM">}}.
+
+Most devices require at least one-half `sck` clock-cycle between the edges of `cs_n` and between either edge of `cs_n` and the nearest `sck` edge.
+However, some devices may require more and thus the SPI_HOST core offers some configuration registers for controlling the timing of the `cs_n` edges when operating under automatic control.
+The relevant parameters are as follows:
+- T<sub>IDLE</sub>: The minimum time between each rising edge of `cs_n` and the following falling edge.
+This time delay is a half `sck` cycle by default but can be extended to as long as four `sck` cycles by setting the {{< regref "CONFIG.CSNIDLE">}} register.
+- T<sub>LEAD</sub>: The minimum time between each falling edge of `cs_n` and the first leading edge of `sck`.
+This time delay is a half `sck` cycle by default but can be extended to as long as four `sck` cycles by setting the {{< regref "CONFIG.CSNLEAD">}} register.
+- T<sub>TRAIL</sub>: The minimum time between the last trailing edge of `sck` and the following rising edge of `cs_n`.
+This time delay is a half `sck` cycle by default but can be extended to as long as four `sck` cycles by setting the {{< regref "CONFIG.CSNTRAIL">}} register.
+
+{{< wavejson >}}
+{signal: [
+  {name: "sck",  wave: "l....1010|10........"},
+  {name: "cs_n", wave: "10.......|.....1...0", node: ".A...B.....C...D...E"}
+],
+ edge: ["A<->B minimum (CSNLEAD+1)", "C<->D minimum (CSNTRAIL+1)", "D<->E minimum (CSNIDLE+1)"],
+  head: {
+    text: "Impact of CSNLEAD, CSNTRAIL and CSNIDLE CONFIG register settings",
+    tick: 1
+  },
+  foot: {
+    text: ["tspan", "All ticks are in units of &#xbd;T",
+           ["tspan", {'baseline-shift':'sub'}, "sck"],
+          "=&#xbd;T",
+           ["tspan", {'baseline-shift':'sub'}, "core"],
+          "&#xd7;(CLKDIV+1)"]
+  }
+}
+{{< /wavejson >}}
+
+These settings are all minimum bounds, and delays in the FSM implementation may create more margin in each of these timing constraints.
+
+### Special Command Fields
+
+The {{< regref "COMMAND" >}} register must be written every time a command is issued.
+Whenever a `1` is written to {{< regref "COMMAND.GO" >}}, contents of the {{< regref "CONFIG" >}} and {{< regref "COMMAND" >}} registers are passed through the Config/Command CDC to the SPI_HOST core FSM.
+Once the command has started, the core will assert {{< regref "STATUS.ACTIVE" >}}.
+The command is complete when {{< regref "STATUS.ACTIVE" >}} goes low.
+An `event` interrupt can also be triggered to go off on completion by setting {{< regref "EVENT_ENABLE.CMD_CMPLT" >}}.
+
+**TODO**: Describe TX1, TXN, dummy, RXN etc.  Refer to previous waveform.
+
+### Chip Select ID's
+
+Each instance of the SPI_HOST IP support up to 8 chip select lines (`cs_n[MaxCS-1:0]`).
```

Comment:
Well, no... we can let MaxCS be less than 8 as things stand now.
The command mechanism just can't support more than 8 with just 3 CSID bits.  Though that brings up a point.  If I let MaxCS !=8, I should signal an error that software asked for a CSID bit that corresponds to an unreachable device.

Reply:
Note to @rswarwick:  I'm thinking of making CONFIG and COMMAND into Multi-regs.  This would eliminate CSID altogether, and would mean that MaxCS could be any size, without this logical inconsistency.

Reply:
CONFIG and COMMAND are now Multi-regs (as of 3f90caeb9439dec510a503a44aeeed92ce1bf55c), and the CSID field is gone (instead it will be inferred based on which mulit-reg gets the `GO` signal.)

Can we consider this point closed?

Reply:
Looks good, thanks!